### DISCUSSION TITLE
v2: Transformer + Packager

### DISCUSSION THREAD

**Discussion by eirikb** [parcel-bundler/parcel#6084] [Q&A] [answered]
Hi

I'm trying to add URL dependency  (`asset.addURLDependency`) to my plugin.  
This works, but the return of `addURLDependency` is just a hash. 
I understand this is meant to be [handled by the packager](https://github.com/parcel-bundler/parcel/discussions/5393).
My plan is to use [@parcel/packager-raw-url](https://www.npmjs.com/package/@parcel/packager-raw-url) to replace hashes with correct file names. Is this a good/correct approach?

I'm asking because I get this error:
> @parcel/packager-raw-url: bundleGraph.resolveExternalDependency is not a function

I could be doing something very wrong. Here is a bare minimum setup I'm using:
https://gist.github.com/eirikb/19f80eb2cadbf0f5dcd96d3d371c54b1
(note; files prefixed *parcel-transformer-eh* are in a subfolder, I don't know how to test plugins beside making a new package).

**Comment by mischnic** [comment, marked as answer]
> Is this the correct approach?

Yes

> bundleGraph.resolveExternalDependency is not a function

I suspect that your version of `@parcel/packager-raw-url` is older than the rest of the packages. I can't find `resolveExternalDependency` anywhere in the current source code for Parcel. Make sure that `parcel` and `@parcel/*` are all on the same version. (i.e. also use the exact version `2.0.0-beta.2` instead of the range `^2.0.0-beta.2`)

**Reply by eirikb** [reply]
That solved it, thanks :)

All I did was to remove ^ and the lock file (lock file was pointing to other versions, so that makes sense).
